segment.go

```diff
@@ -247,18 +218,82 @@ func (seg *segment) Write(data []byte) (*ChunkPosition, error) {
 		default: // Middle chunk
 			chunkType = ChunkTypeMiddle
 		}
-		seg.appendChunkBuffer(chunkBuffer, data[dataSize-leftSize:end], chunkType)
+		seg.appendChunkBuffer(buf, data[dataSize-leftSize:end], chunkType)
 
 		leftSize -= chunkSize
 		blockCount += 1
 		currBlockSize = (currBlockSize + chunkSize + chunkHeaderSize) % blockSize
 	}
 
+	position.ChunkSize = blockCount*chunkHeaderSize + dataSize
+
+	return position, padding, nil
+}
+
+// WriteAll write batch data to the segment file.
+func (seg *segment) WriteAll(data [][]byte) ([]*ChunkPosition, error) {
+	if seg.closed {
+		return nil, ErrClosed
+	}
+
+	// init chunk buffer
+	chunkBuffer := bytebufferpool.Get()
+	defer func() {
+		chunkBuffer.Reset()
+		bytebufferpool.Put(chunkBuffer)
+	}()
+
+	positions := make([]*ChunkPosition, len(data))
+	currentBlockNumber := seg.currentBlockNumber
+	currentBlockSize := seg.currentBlockSize
+	for i := 0; i < len(positions); i++ {
+		pos, padding, err := seg.calChunkPosition(data[i], chunkBuffer, currentBlockNumber, currentBlockSize)
+		if err != nil {
+			return nil, err
+		}
+		currentBlockSize += padding + pos.ChunkSize
+		if currentBlockSize >= blockSize {
+			currentBlockNumber += currentBlockSize / blockSize
+			currentBlockSize = currentBlockSize % blockSize
+		}
+		if err != nil {
+			return nil, err
+		}
+		positions[i] = pos
+	}
+	if err := seg.writeChunkBuffer(chunkBuffer); err != nil {
+		return nil, err
+	}
+	return positions, nil
+}
+
+// Write writes the data to the segment file.
+// The data will be written in chunks, and the chunk has four types:
+// ChunkTypeFull, ChunkTypeFirst, ChunkTypeMiddle, ChunkTypeLast.
+//
+// Each chunk has a header, and the header contains the length, type and checksum.
+// And the payload of the chunk is the real data you want to write.
+func (seg *segment) Write(data []byte) (*ChunkPosition, error) {
+	if seg.closed {
+		return nil, ErrClosed
+	}
+	// init chunk buffer
+	chunkBuffer := bytebufferpool.Get()
```

Comment:
i got rid of the bytebufferpool and used a sole buffer embedded to the segment because the access to this buffer is always synchronos so i saw no point in that. lemme know what u think